ToolBox/GoogleDownloader/GoogleDataDownloader.cs

```diff
@@ -32,7 +32,7 @@ public class GoogleDataDownloader : IDataDownloader
         // q = SYMBOL
         // startdate = Mon+D%2C+YYYY, e.g. Jan+1%2C+2005
         // enddate = Mon+D%2C+YYYY, e.g. Sep+29%2C+2016
-        private const string UrlPrototypeDaily = @"http://www.google.com/finance/historical?q={0}&startdate={1}&enddate={2}&output=csv";
+        private const string UrlPrototypeDaily = @"https://finance.google.com/finance/historical?q={0}&startdate={1}&enddate={2}&output=csv";
```

Comment:
Is this correct?

I created both urls and only the old one worked for me:
```
// works (old)
http://www.google.com/finance/historical?q=SPY&startdate=20170101&enddate=20180101&output=csv

// does not work (new)
http://www.finance.google.com/finance/historical?q=SPY&startdate=20170101&enddate=20180101&output=csv
```

Reply:
Those URLs don't look correct in the first place unless Google Finance accepts the date encoded as `yyyymmdd`. What worked for me is the following…

## Old

The URL is:

> http://www.google.com/finance/historical?q=MSFT&startdate=May+25%2C+2005&enddate=Oct+10%2C+2017&output=csv

You can test using cURL as follows:

```
curl -vs --location 'http://www.google.com/finance/historical?q=MSFT&startdate=May+25%2C+2005&enddate=Oct+10%2C+2017&output=csv' 1>old.txt 2>&1
```

Note that [`old.txt`](https://github.com/QuantConnect/Lean/files/1380707/old.txt) in the end contains a redirection:

```
< HTTP/1.1 302 Found
< Location: http://finance.google.com/finance/historical?q=MSFT&output=csv
```

The file weighs in at 13,086 bytes. Your mileage may vary slightly.

## New

The URL is:

> https://finance.google.com/finance/historical?q=MSFT&startdate=May+25%2C+2005&enddate=Oct+10%2C+2017&output=csv

```
curl -vs --location 'https://finance.google.com/finance/historical?q=MSFT&startdate=May+25%2C+2005&enddate=Oct+10%2C+2017&output=csv' 1>new.txt 2>&1
```

The [`new.txt`](https://github.com/QuantConnect/Lean/files/1380721/new.txt)
 file weighs in at 134,744 bytes (again, you mileage may vary slightly) and is much bigger because it contains the entire request history.

## Other Information

I used cURL on macOS. Below is the version information as emitted by `curl -V`:

```
curl 7.54.0 (x86_64-apple-darwin16.0) libcurl/7.54.0 SecureTransport zlib/1.2.8
Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps pop3 pop3s rtsp smb smbs smtp smtps telnet tftp 
Features: AsynchDNS IPv6 Largefile GSS-API Kerberos SPNEGO NTLM NTLM_WB SSL libz UnixSockets 
```

Reply:
@atifaziz - I greatly appreciate the thoroughness. It looks like I overlooked the format google accepts for the date. Interestingly it accepts `yyyyMMdd` in the other url.